lib/DownloadManager.js

```diff
@@ -85,8 +86,9 @@ class DownloadManager {
             }
             var fulfill;
             var promise = new Promise(x => fulfill = x);
-            logger.info(`git clone ${url} ${folderPath}`);
-            exec('git', ['clone', '--depth', '1', url, folderPath], {timeout: utils.minutes(2)}, (err, stdout, stderr) => {
+            var args = ['clone', '--depth', '1', '-b', revision, url, folderPath];
```

Comment:
@NonlyNerd this doesn't seem to work for me:

```
$ git clone https://github.com/aslushnikov/latex-online --depth 1 -b 18b2131a29b1946e0180e2f58493eef988b2950d
Cloning into 'latex-online'...
warning: Could not find remote branch 18b2131a29b1946e0180e2f58493eef988b2950d to clone.
fatal: Remote branch 18b2131a29b1946e0180e2f58493eef988b2950d not found in upstream origin
```

Reply:
My bad. I forgot the shallow thing and the handling of git-clone for revision ( I mixed up checkout and clone). For checking with a commit identifier, we have to work with a full clone, not a shallow one. I've updated the README to not disrupt users on the usage.